src/server/actions/Setting.ts

```diff
@@ -0,0 +1,33 @@
+import SettingsSchema from '@/server/models/Settings'
+import UserSchema from '@/server/models/User'
+import { SettingsResponse } from '@/types/dataModel/settings'
+import CMError, { CMErrorType } from '@/utils/cmerror';
+import dbConnect from '@/utils/db-connect'
+import { getSettings } from './Settings';
+
+export async function removeAllowedEmail(email: string) {
+    let doc: SettingsResponse
+    try {
+        dbConnect();
+        doc = await getSettings()
+    } catch (error) {
+        throw new CMError(CMErrorType.InternalError)
+    }
+
+    const index = doc.allowedEmails.indexOf(email)
+    // No matching email found in email document
+    if (index == -1) {
+        throw new CMError(CMErrorType.NoSuchKey, "Email")
+    } else {
+        // Modifies array in-place
+        doc.allowedEmails.splice(index, 1)
+        // Set database obj array to modified array of retrieved document
+        await SettingsSchema.updateOne({ _id: doc._id }, { allowedEmails: doc.allowedEmails })
+
+        // Delete associated user if one exists
+        try {
+            await UserSchema.findOneAndRemove({ email: email })
+        } catch (e) { }
```

Comment:
why are there two try catches here?

Reply:
First try catch is for throwing error if database call fails. Second try-catch never throws an error, it's just necessary to delete a user with the given email if one exists and to do nothing if one doesn't exist.